codalab/server/bundle_manager.py

```diff
@@ -339,11 +332,13 @@ def _deduct_worker_resources(self, workers_list):
         for worker in workers_list:
             for uuid in worker['run_uuids']:
                 bundle = self._model.get_bundle(uuid)
-                worker['cpus'] -= self._compute_request_cpus(bundle)
-                worker['gpus'] -= self._compute_request_gpus(bundle)
-                worker['memory_bytes'] -= self._compute_request_memory(bundle)
+                bundle_resources = self._compute_bundle_resources(bundle)
```

Comment:
While this adds a line of code I think it's just better to have method that does all resource computations rather than having to call individual functions for individual fields every time we need them.